templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,9 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       var selected_status_id  = obj.find(":selected").val();
+       var cancelInfo_show_ids = [{/literal}{$cancelInfo_show_ids}{literal}];
+       if (cancelInfo_show_ids.indexOf(selected_status_id) > -1) {
```

Comment:
hmm, it looks like you are searching a string and not an array. So if the string is

    8,12

And you search for the value "2" it will return `true`.

I suggest that when you assign the variable to the template you use `json_encode` rather than `implode`.